packages/vue/src/components/list/SingleList.jsx

```diff
@@ -132,6 +121,17 @@ const SingleList = {
 		this.removeComponent(this.internalComponent);
 	},
 	watch: {
+		$props: {
+			deep: true,
+			handler(newVal) {
+				this.updateComponentProps(this.componentId, newVal, componentTypes.singleList);
```

Comment:
@ShahAnuj2610 What is `newVal` here?
Can you log it and share a video?
Also for the performance reasons only update the valid props.

Reply:
@bietkul 
> What is newVal here?

it's the new value for props. Will share the video.
> Also for the performance reasons only update the valid props

it's already handled in `updateComponentProps`
https://github.com/appbaseio/reactivecore/blob/c6deb6e40da8da8eb2ab7b1397ce838098c61e58/src/actions/props.js#L35
https://github.com/appbaseio/reactivecore/blob/c6deb6e40da8da8eb2ab7b1397ce838098c61e58/src/actions/props.js#L4